File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxiprofit = 0;
        int buy = prices[0];
        for(int i = 0; i<prices.size() ; i++){
            if(buy > prices[i]){
                buy = prices[i];
            }else{
                int currprofit = prices[i] - buy;
                maxiprofit = max(maxiprofit , currprofit);
            }
        }
        return maxiprofit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]
        maxprofit = 0
        for i in range(len(prices)):
            if prices[i] < buy:
                buy = prices[i]
            else:
                currentprofit = prices[i] - buy
                maxprofit = max(maxprofit , currentprofit)
        return maxprofit
